test/integration/dbTest/getHistoricAlertsByAlertApiKeyTest.js

```diff
@@ -0,0 +1,386 @@
+          responded_at: this.respondedAt,
+        },
+      ])
+    })
+  })
+
+  describe('if there are more matching sessions than maxHistoricAlerts', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearLocations()
+
+      // Insert a single location and more than maxHistoricAlerts sessions
+      this.alertApiKey = 'alertApiKey'
+      const locationid = 'locationid'
+      await db.createLocation(
+        locationid,
+        'phonenumber',
+        'movementThreshold',
+        'stillnessThreshold',
+        'durationTimer',
+        120000,
+        'initialTimer',
+        '{"heartbeatAlertRecipients"}',
+        'twilioNumber',
+        '{"fallbackNumbers"}',
+        300000,
+        'displayName',
+        'doorCoreId',
+        'radarCoreId',
+        'radarType',
+        this.alertApiKey,
+        true,
+        false,
+      )
+
+      this.session1 = await db.createSession(locationid, 'phonenumber1', ALERT_TYPE.SENSOR_DURATION)
+      this.session2 = await db.createSession(locationid, 'phonenumber2', ALERT_TYPE.SENSOR_DURATION)
+      this.session3 = await db.createSession(locationid, 'phonenumber3', ALERT_TYPE.SENSOR_DURATION)
+      this.session4 = await db.createSession(locationid, 'phonenumber4', ALERT_TYPE.SENSOR_DURATION)
+      this.session5 = await db.createSession(locationid, 'phonenumber5', ALERT_TYPE.SENSOR_DURATION)
+    })
+
+    afterEach(async () => {
+      await db.clearSessions()
+      await db.clearLocations()
+    })
+
+    it('should return only the most recent maxHistoricAlerts of them', async () => {
+      const rows = await db.getHistoricAlertsByAlertApiKey(this.alertApiKey, 3, 1)
+
+      const ids = rows.map(row => row.id)
+
+      expect(ids).to.eql([this.session5.id, this.session4.id, this.session3.id])
+    })
+  })
+
+  describe('if there are fewer matching sessions than maxHistoricAlerts', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearLocations()
+
+      // Insert a single location and maxHistoricAlerts sessions
+      this.alertApiKey = 'alertApiKey'
+      const locationid = 'locationid'
+      await db.createLocation(
+        locationid,
+        'phonenumber',
+        'movementThreshold',
+        'stillnessThreshold',
+        'durationTimer',
+        120000,
+        'initialTimer',
+        '{"heartbeatAlertRecipients"}',
+        'twilioNumber',
+        '{"fallbackNumbers"}',
+        300000,
+        'displayName',
+        'doorCoreId',
+        'radarCoreId',
+        'radarType',
+        this.alertApiKey,
+        true,
+        false,
+      )
+
+      this.session1 = await db.createSession(locationid, 'phonenumber1', ALERT_TYPE.SENSOR_DURATION)
+      this.session2 = await db.createSession(locationid, 'phonenumber2', ALERT_TYPE.SENSOR_DURATION)
+      await helpers.sleep(2000) // Couldn't get around doing this because there is a trigger on updated_at to keep it correct
```

Comment:
does the 2000 refer to SESSION_RESET_THRESHOLD_TEST in travis.yml? if so chould we use an env var instead to make this test less likely to flake?

Reply:
No, it was just to make sure that session1 and session2 happen after the `1000` ms passed into `const rows = await db.getHistoricAlertsByAlertApiKey(this.alertApiKey, 5, 1000)` in the test. But I should actually be able to reduce it to 1000 to match that value and that will work too. I can use a variable to make that association more obvious. I haven't seen any flakiness with the 1000ms, but if our DB connection is slow for some reason, then I could see us needing to increase it eventually.